src/main/java/net/minecraftforge/items/pickup/ItemPickupPredicate.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.items.pickup;
+
+import org.jetbrains.annotations.NotNull;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.item.ItemEntity;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+
+import java.util.Set;
+
+/**
+ * FunctionalInterface used to test the Level to determine if the specified ItemEntity is valid to be picked up or not.
+ *
+ * <p>Any predicate which returns false will disallow the item from being picked up.
+ * <p>All predicates must return true, in order for the item to be picked up by the collector.
+ */
+@FunctionalInterface
+public interface ItemPickupPredicate
+{
+    /**
+     * Test if the specified ItemEntity is valid to be picked up or not.
+     *
+     * @param item The ItemEntity being picked up
+     * @param stack The ItemStack being picked up
+     * @param level Level containing the ItemEntity
+     * @param pos Position of the ItemEntity
```

Comment:
The item stack, level, and position are all redundant as the Item entity contains getters for all of these. Allowing them to be passed in allows for the opportunity for disagreements.